src/diffusers/models/model_loading_utils.py

```diff
@@ -176,11 +175,9 @@ def load_model_dict_into_meta(
     hf_quantizer=None,
     keep_in_fp32_modules=None,
 ) -> List[str]:
-    if hf_quantizer is None:
-        device = device or torch.device("cpu")
+    device = device or torch.device("cpu")
```

Comment:
Why change what we already have been using?

Reply:
The type hint annotates this to only accept str or torch.device. I think that is incorrect then (we can fix it in this PR). Could we change `param_device = torch.cuda.current_device()` to `param_device = torch.device(torch.cuda.current_device())` instead in modeling_utils.py?

Reply:
Yeah that should work. Would prefer that in a separate PR and merge quickly before this one so that it can be rigorously tested first (I will help). Works?

Reply:
sounds good, i'll open shortly!

Reply:
Thank you https://github.com/huggingface/diffusers/pull/10069